### PR TITLE
Update ubuntu-latest to use Ubuntu-24.04

### PR DESCRIPTION
**Pull request by Bryan-Roe**
Fixes #10636

Update workflows and documentation to use Ubuntu-24.04 as the default image for `ubuntu-latest`.

* **README.md**
  - Update the `ubuntu-latest` label to point to Ubuntu-24.04.
  - Remove the `beta` tag from the `ubuntu-24.04` image.

* **Workflows**
  - Update `.github/workflows/codeql-analysis.yml`, `.github/workflows/create_github_release.yml`, `.github/workflows/create_pull_request.yml`, `.github/workflows/linter.yml`, `.github/workflows/merge_pull_request.yml`, `.github/workflows/powershell-tests.yml`, `.github/workflows/trigger-ubuntu-win-build.yml`, and `.github/workflows/update_github_release.yml` to use `runs-on: ubuntu-24.04`.

* **Ubuntu2404-Readme.md**
  - Confirm the rollout completion of Ubuntu-24.04 as `ubuntu-latest`.

---

For more details, open the [Copilot Workspace session](https://copilot-workspace.githubnext.com/actions/runner-images/pull/11131?shareId=c120198a-3ce8-470e-a482-802805fcf3cd).

### CODE REVIEW

**Review by Kichura — COMMENTED** [review]
Can't the `ubuntu-latest` sections remain unchanged as then 24.04 would take priority or what is the reasoning behind this one?

**Review by FreeMason9224 — COMMENTED** [review]
👍

**Review by invisiblepancake — APPROVED** [review]

**Review by invisiblepancake — APPROVED** [review]

File: .devcontainer/devcontainer.json
Comment (TheGiraffe3) [inline .devcontainer/devcontainer.json:1]: This looks like an unrelated change. What is it for?

**Review by Jenibabi1984 — APPROVED** [review]

File: .github/workflows/powershell-tests.yml [hunk 1]
```diff
@@ -22,4 +22,4 @@ jobs:
         run: |
             $ErrorActionPreference = "Stop"
             Invoke-Pester -Output Detailed "helpers/software-report-base/tests"
-        
\ No newline at end of file
+
```
Comment (TheGiraffe3) [inline .github/workflows/powershell-tests.yml:25]:
```suggestion

```
whitespace removal

**Comment by abishakla** [comment]
aqua
aqua

**Comment by TRIMIXING1AI** [comment]
gh pr checkout 11131

**Review by TRIMIXING1AI — APPROVED** [review]
images/ubuntu/Ubuntu2404-Readme.md

**Review by TRIMIXING1AI — COMMENTED** [review]
.github/workflows/codeql-analysis.yml

**Review by Mrvp215 — APPROVED** [review]

**Review by invisiblepancake — COMMENTED** [review]
lets try

**Review by MrG9090 — APPROVED** [review]

**Review by Solo2020vtc — APPROVED** [review]

**Review by MrG9090 — APPROVED** [review]

**Review by Bryan-Roe — COMMENTED** [review]
Retrying

**Review by MrG9090 — APPROVED** [review]

**Comment by RFOF-NETWORK** [comment]
What cind of support do you need  & whats the labels from this task when you can clear my questions than still response me. 

Ps: so fast as you can !

**Review by MrG9090 — APPROVED** [review]

**Review by Wbaker7702 — COMMENTED** [review]
Main
